File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    
    var isFind = false;
    let loc_val = priorities[location];
    priorities.forEach((value, idx) => {
        priorities[idx] = [value, idx]
        
    })
    while (!isFind) {
        let maxPrior = -1;
        priorities.forEach((value, idx) => {
            maxPrior = Math.max(maxPrior, value[0])        
        })
        let target = priorities.shift()

        if (target[0] == maxPrior) {
            answer ++;
            if (target[1] == location) {
                isFind = true;
            }
        } else {
            priorities.push([target[0], target[1]]);
        }
    }
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    queue = deque()
    for i in range(len(priorities)):
        queue.append((i, priorities[i]))
        
    while queue:
        idx, prior = queue.popleft()
        print(idx, prior)
        maxPrior = -1
        for i in range(len(queue)):
            t, p = queue[i]
            maxPrior = max(maxPrior, p)
        
        if prior >= maxPrior:
            answer += 1
            if idx == location:
                return answer
        else:
            queue.append((idx, prior))
        
    
    return answer